lib/nrf_modem_lib/nrf91_sockets.c

```diff
@@ -1100,4 +1114,18 @@ NET_DEVICE_OFFLOAD_INIT(nrf91_socket, "nrf91_socket",
 			&nrf91_iface_data, NULL,
 			0, &nrf91_iface_offload_api, 1280);
 
-#endif
+#if defined(CONFIG_LTE_CONNECTIVITY)
+/* Bind l2 connectity APIs. */
+static struct conn_mgr_conn_api conn_api = {
+	.init = lte_connectivity_init,
+	.connect = lte_connectivity_connect,
+	.disconnect = lte_connectivity_disconnect,
+	.set_opt = lte_connectivity_options_set,
+	.get_opt = lte_connectivity_options_get,
+};
+
+CONN_MGR_CONN_DEFINE(LTE_CONNECTIVITY, &conn_api);
+CONN_MGR_BIND_CONN(nrf91_socket, LTE_CONNECTIVITY);
```

Comment:
side comment: We should probably come up with a better name for the lte interface than `nrf91_socket` at some point